File: lib/features/wedding/data/models/task_model.dart
import 'package:wedding_witness_app/features/wedding/domain/entities/task.dart';

class TaskModel extends Task {
  TaskModel({
    required super.id,
    super.title,
    super.description,
    super.startTime,
    super.isCompleted,
    super.songId,
  });

  // Méthode pour convertir un JSON en TaskModel
  factory TaskModel.fromJson(Map<String, dynamic> json) {
    return TaskModel(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      startTime: json['startTime'],
      isCompleted: json['isCompleted'] ?? false,
      songId: (json['songId'] as int?) == 0 ? null : json['songId'] as int?,
    );
  }

  // Méthode pour convertir TaskModel en Map (pour la persistance)
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'isCompleted': isCompleted,
      'description': description,
      'startTime': startTime,
      'songId': songId,
    };
  }
}

File: server/app/infrastructure/models/task.model.go
package models

type TaskModel struct {
	ID          uint   `gorm:"primaryKey" json:"id"`
	Title       string `json:"title"`
	Description string `json:"description"`
	IsCompleted bool   `json:"is_completed" gorm:"default:false"`
	StartTime   string `json:"start_time"`
	SongID      uint   `json:"song_id" gorm:"default:0"`
}
